xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/script/RequestFactory.java

```diff
@@ -153,7 +154,17 @@ public CopyRequest createCopyRequest(Collection<EntityReference> sources, Entity
     {
         CopyRequest request = new CopyRequest();
         this.initEntityRequest(request, RefactoringJobs.COPY, sources);
-        request.setDestination(destination);
+
+        EntityReference dest;
+
+        // in case of a simple copy request, we only want it to be placed under the given destination document.
+        if (destination.getType().equals(EntityType.DOCUMENT)) {
+            DocumentReference documentReference = (DocumentReference) destination;
```

Comment:
I'm not sure you could do this cast. I think we usually write:
```suggestion
            DocumentReference documentReference = new DocumentReference(destination);
```

Reply:
thanks for your feedback!